mdtraj/formats/pdb/pdbfile.py

```diff
@@ -406,11 +406,12 @@ def _write_footer(self):
                             'A', 'G', 'C', 'U', 'I', 'DA', 'DG', 'DC', 'DT', 'DI', 'HOH']
         conectBonds = []
         if self._last_topology is not None:
+            atoms = list(self._last_topology.atoms)
```

Comment:
There is likely a more sophisticated way of solving this. It seems that the atoms stored in `topology.bonds` differ by the chain index (mdtraj/core/topology.py::1569) from the atoms retrieved by iterating through chains. Unclear as what is causing that behavior based on a initial read of the code.